foundation/foundation-upgrade/src/main/java/com/boozallen/aissemble/upgrade/migration/v1_11_0/SparkInfrastructureUniversalConfigYAMLMigration.java

```diff
@@ -0,0 +1,112 @@
+        try {
+            YamlUtils.YamlObject yaml = YamlUtils.loadYaml(file);
+            if(yaml.hasObject("aissemble-hive-metastore-service-chart"))
+            {
+                yaml = yaml.getObject("aissemble-hive-metastore-service-chart");
+            }else{
+                return false;
+            }
+            boolean hasMySqlAuthDefaultUsername = false;
+            if (yaml.hasObject("mysql", "auth")) {
+                YamlUtils.YamlObject mysqlAuth = yaml.getObject("mysql", "auth");
+                for(String key : mysqlAuth.keySet())
+                {
+                    if(mysqlAuthForConfigStore.contains(key) && mysqlAuth.get(key).equals(DEFAULT_METASTORE_DB_USERNAME))
+                    {
+                        hasMySqlAuthDefaultUsername = true;
+                    }
+                }
+
+            }
+            boolean hasMetastoreServiceConfigDefaultUsername = false;
+            if (yaml.hasList("configMap", "metastoreServiceConfig", "properties")) {
+                List<YamlUtils.YamlObject> properties = yaml.getListOfObjects("configMap", "metastoreServiceConfig", "properties");
+                for(YamlUtils.YamlObject property : properties)
+                {
+                    if(propertiesForConfigStore.contains(property.get("name")) && property.get("value").equals(DEFAULT_METASTORE_DB_USERNAME))
+                    {
+                        hasMetastoreServiceConfigDefaultUsername = true;
+                        break;
+                    }
+                   }
+            }
+            return hasMySqlAuthDefaultUsername && hasMetastoreServiceConfigDefaultUsername;
+
+        } catch (YAMLException e) {
+            if (logger.isDebugEnabled()) {
+                logger.debug("Failed to parse values yaml file: {}", file.getPath(), e);
+            } else {
+                logger.info("Failed to parse values yaml file: {}", file.getName());
+            }
+        }
+        catch (IOException e) {
+            throw new RuntimeException(e);
+        }
+        return false;
+    }
+
+    @Override
+    protected boolean performMigration(File file) {
+        try {
+            File tempFile = new File("tempValuesFile.yaml");
+            BufferedWriter writer = new BufferedWriter(new FileWriter(tempFile));
+            List<String> lines = Files.readAllLines(file.toPath());
+            for (int i = 0; i < lines.size(); i++) {
+                String line = lines.get(i).trim();
+                boolean skipConnectionPassword = false;
+                if(i >= 1 && lines.get(i-1).trim().equals("- name: javax.jdo.option.ConnectionPassword"))
+                {
+                    skipConnectionPassword = true;
+                }
+                if(linesToRemove.contains(line) && !skipConnectionPassword)
+                {
+                    continue;
+                }
+                writer.write(lines.get(i) + System.getProperty("line.separator"));
```

Comment:
I believe this logic will break in the event the password appears before the username:
```
        - name: javax.jdo.option.ConnectionPassword
          value: hive
          description: Password to use against metastore database
        - name: javax.jdo.option.ConnectionUserName
          value: hive
          description: Username to use against metastore database
```

Think it could be resolved by updating the logic to the following:
```
                if(linesToRemove.contains(line))
                {
                    if(!skipConnectionPassword) {
                        continue;
                   } else {
                       skipConnectionPassword = false;
                   }
                }
```

Reply:
Hmm not sure my logic fails with the case you mentioned,

what I am checking for is if name is javax.jdo.option.ConnectionPassword (which is previous line i-1)we set skipConnectionPassword to true and skip writing file but after that iteration skipConnectionPassword defaults back to false on every line so I think this logic works

 I did try unit test with password coming first and it still passes..

Reply:
Oh you're right, didn't realize the boolean resets each iteration